src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java

```diff
@@ -664,12 +664,12 @@ public void addModAsResource(ModContainer container)
             catch (NoSuchMethodException e)
             {
                 FMLLog.log.error("The container {} (type {}) returned an invalid class for it's resource pack.", container.getName(), container.getClass().getName());
-                return;
             }
             catch (Exception e)
             {
                 FMLLog.log.error("An unexpected exception occurred constructing the custom resource pack for {}", container.getName(), e);
-                throw Throwables.propagate(e);
+                Throwables.throwIfUnchecked(e);
+                throw new RuntimeException(e);
```

Comment:
I think that any place where we log an error before rethrowing a caught exception should be written a different way.

We should be putting the message with the exception, so it shows up in the exception "caused by" chain instead of being hidden away in the log somewhere before the crash.

For example, this one should be:
```java
throw new RuntimeException("An unexpected exception occurred constructing the custom resource pack for " + container.getName(), e);
```

Can you update these? If not, I can go through and update them after this is merged.